File: exec/src/encapsulamento_1/Acesso_Numeros.java
package encapsulamento_1;

public class Acesso_Numeros {
    int a;
    public int b;
    private int c;
    protected int d;

    public void setNumero(String id, int numero) {
        if(id=="a")
            this.a = numero;

        if(id=="c")
            this.c = numero;

        if(id=="d")
            this.d = numero;

    }

    public void mostra_numero() {
        System.out.println("numero a =" + a);
        System.out.println("numero b =" + b);
        System.out.println("numero c =" + c);
        System.out.println("numero d =" + d);

    }
}

File: comparacao_python/encapsulamento_1/Acesso_Numeros.py
class Acesso_Numeros:
    def __init__(self):
        self.a = 0  # Público
        self.b = 0  # Público
        self.__c = 0  # Privado
        self._d = 0  # Protected

    def set_numero(self, id, numero):
        if id == "a":
            self.a = numero
        elif id == "c":
            self.__c = numero
        elif id == "d":
            self._d = numero

    def mostra_numero(self):
        print(f"numero a = {self.a}")  # Acesso ao atributo público 'a'
        print(f"numero b = {self.b}")  # Acesso ao atributo público 'b'
        print(f"numero c = {self.__c}")  # Acesso ao atributo protegido '_c'
        print(f"numero d = {self._d}")  # Acesso ao atributo privado '__d'